internal/app/app.go

```diff
@@ -129,25 +133,64 @@ func Run(ctx context.Context, osArgs []string, h *common.Health, optLoggerConfig
 
 	logger.Infoln("starting workers...")
 
-	h.AddReadiness()
-	eg.Go(func() error {
-		jp := journald.Processor{
-			BootID:    bootID,
-			MachineID: mid,
-			NodeName:  nodename,
-			Distro:    distro,
-			EventW:    eventWriter,
-			Logins:    logins,
-			CurrentTS: lastReadJournalTS,
-			Health:    h,
-		}
-
-		err := jp.Read(groupCtx)
-		if logger.Level().Enabled(zap.DebugLevel) {
-			logger.Debugf("journald worker exited (%v)", err)
-		}
-		return err
-	})
+	if distro == util.DistroRocky {
+		eg.Go(func() error {
+			// Create a tail
+			t, err := tail.TailFile(
```

Comment:
pulled a go tail library for tailing the `/var/log/secure` file

Reply:
I reviewed the library. It looks good. Seems to get the job done. if we see any errors we can easily replace the library with something else. Also didn't want to spend alot of time refactoring the `rotateFile` code to public since we may end up using the named pipe abstraction.